### ISSUE TITLE
Example command w/ wallhaven.cc triggers crash

### ISSUE THREAD

**Issue by verbunk** [jkotra/wpc#10]
Calling  ./wpc -d . -w triggers a crash

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/web/wallhaven.rs:96:34
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1069
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1504
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:511
  11: rust_begin_unwind
             at src/libstd/panicking.rs:419
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:111
  13: core::option::expect_none_failed
             at src/libcore/option.rs:1268
  14: wpc::wallhaven::WallHaven::read_json
  15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  16: tokio::park::thread::CachedParkThread::block_on
  17: tokio::runtime::thread_pool::ThreadPool::block_on
  18: tokio::runtime::Runtime::block_on
  19: wpc::main
  20: std::rt::lang_start::{{closure}}
  21: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  22: std::panicking::try::do_call
             at src/libstd/panicking.rs:331
  23: std::panicking::try
             at src/libstd/panicking.rs:274
  24: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  25: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  26: main
  27: __libc_start_main
  28: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

System : 
- Artix Linux
- 64GB / 16 cores

#> rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/verbunk/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
1.41.0-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.47.0 (18bf6b4f0 2020-10-07)

**Comment by verbunk** [comment]
Forgot wpc stats,

./wpc --help
WPC 1.0.0
Jagadeesh K. <jagadeesh@stdin.top>
Wallpaper changer for Windows/Linux

**Comment by jkotra** [comment]
this has been fixed in 1.0.1:
https://github.com/jkotra/wpc/releases/tag/1.0.1

**Comment by verbunk** [comment]
Hey Jagadeesh, 

Github steered me to 1.0.0 as latest release but I did see 1.0.1 after you pointed it out. Still seems there is an issue. Error message changed slightly but is largely the same.

```
$> RUST_BACKTRACE=1 ./wpc -d . -w
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/web/wallhaven.rs:100:34
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1069
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1504
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:511
  11: rust_begin_unwind
             at src/libstd/panicking.rs:419
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:111
  13: core::option::expect_none_failed
             at src/libcore/option.rs:1268
  14: wpc::wallhaven::WallHaven::read_json
  15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  16: tokio::park::thread::CachedParkThread::block_on
  17: tokio::runtime::thread_pool::ThreadPool::block_on
  18: tokio::runtime::Runtime::block_on
  19: wpc::main
  20: std::rt::lang_start::{{closure}}
  21: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  22: std::panicking::try::do_call
             at src/libstd/panicking.rs:331
  23: std::panicking::try
             at src/libstd/panicking.rs:274
  24: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  25: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  26: main
  27: __libc_start_main
  28: _start

```

**Comment by jkotra** [comment]
Hi @verbunk, thanks for reporting the bug :+1: .

This has been fixed in `1.0.2`
https://github.com/jkotra/wpc/releases/tag/1.0.2
